interface/grpc/core/create_workflow.go

```diff
@@ -0,0 +1,22 @@
+package core
+
+import (
+	"context"
+
+	"github.com/mesg-foundation/core/protobuf/coreapi"
+	"github.com/mesg-foundation/core/systemservices/workflow"
+)
+
+// CreateWorkflow creates and runs a new workflow.
+func (s *Server) CreateWorkflow(ctx context.Context, request *coreapi.CreateWorkflowRequest) (*coreapi.CreateWorkflowReply, error) {
+	exec, err := s.api.ExecuteAndListen(s.ss.WorkflowServiceID(), workflow.CreateTaskKey, workflow.CreateInputs(request.File, request.Name))
```

Comment:
Could you provide an example about how to move this piece of code to _core/api_ please? You can describe the new funcs that we need to create in _core/api_ with their arguments and return values. No need for an actual implementation in the funcs' body.

Reply:
Basically it would be something like that (this is not tested and just done quickly to get the idea)

## package `api`
```go
package api

func (a *API) CreateWorkflow(workflowDefinition WorkflowDefinition) (workflowID string, err error) {
	return newWorkflowCreator(a, ss).Create(workflowDefinition)
}

type workflowCreator struct {
	api *API
	ss *systemservices.SystemServices
}

func newWorkflowCreator(api *API, ss *systemservices.SystemServices) *workflowCreator {
	return &workflowCreator{
		api: api,
		ss: ss,
	}
}

func (w *workflowCreator) Create(workflowDefinition WorkflowDefinition) (workflowID string, err error) {
	exec, err := w.api.ExecuteAndListen(w.ss.WorkflowServiceID(), "create", workflow.CreateInputs(definition))
	if err != nil {
		return nil, err
	}
 	return workflow.CreateOutputs(exec)
}
```

## package `interface/grpc/core`

```go
package core

func (s *Server) CreateWorkflow(ctx context.Context, request *coreapi.CreateWorkflowRequest) (*coreapi.CreateWorkflowReply, error) {
	id, err := s.api.CreateWorkflow(request.WorkflowDefinition)
	if err != nil {
		return nil, err
	}
	return &coreapi.CreateWorkflowReply{
		id: id,
	}, 
}
```

Reply:
This is great, I had the same idea but we can't import _systemservices_ inside _api_ because it's [cycling](https://github.com/mesg-foundation/core/blob/62ae275/systemservices/systemservices.go#L6).

I think this is relevant with the second bullet you mentioned in your [comment](https://github.com/mesg-foundation/core/pull/535#issuecomment-436565176) in system services PR. I'd like to know what kind of structure you're thinking to create in the _systemservices_ & _api_ pkgs while moving service deploy/start logic to _api_ pkg.

Reply:
we discussed about this one on the weekly meeting and we decided to create a new pkg under named _systemservices/deployer_ and it'll be responsible to deploy and start system services. _systemservices_ pkg will call method(s) from this pkg in the initialization process.

Reply:
originally related with https://forum.mesg.com/t/the-import-cycle-between-api-systemservices-packages/105

Reply:
solved by https://github.com/mesg-foundation/core/pull/567